grype/vulnerability/provider.go

```diff
@@ -13,8 +13,10 @@ type Criteria interface {
 }
 
 // MetadataProvider implementations provide ways to look up vulnerability metadata
+// Deprecated: vulnerability.Vulnerability objects now have metadata included
```

Comment:
I think we should at least remove `MetadataProvider` from the provider, below, if possible

Reply:
I think that can be done -- but in a follow up PR (or part of grype 1.0 work)